docs/js/features/odata/generate-client.md

```diff
@@ -0,0 +1,143 @@
+---
+id: generator-js-sdk
+title: Generate an OData client for JavaScript
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Generate an OData client for JavaScript
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- JavaScript 
+- TypeScript
+- generate
+author: Charles Dubois
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Generate a typed OData client with the SAP Cloud SDK generator
+
+The generator allows you to generate custom OData client libraries for OData services. You can then easily access these services via the client libraries.
+
+The SAP Cloud SDK generator can both be used as a command line interface (CLI) and programmatically.
+
+All you need to use it is a service metadata specification in the `EDMX` format (file ending can be `.edmx` or `.xml`).
+
+## Installation
+
+Run this command in your project's terminal to install the generator globally.
+```sh
+npm install -g @sap-cloud-sdk/generator
+```
+
+:::note
+The generator is included in the SAP Cloud SDK CLI. If you installed the CLI, you don't need to install the generator separately.
+:::
+
+## Using the OData Generator
+
+### Using the Command Line Interface
+
+The SAP Cloud SDK generator is primarily intended to be used on the command line.
+
+Run
+```shell
+generate-odata-client --inputDir path/to/your/service-specifications --outputDir path/to/store/generated/modules
+```
+
+Adapt the `path/to/your/service-specifications` to the directory containing your service specifications in `EDMX` format, for example `service-specifications/`. Also adapt `path/to/store/generated/modules` to your OData client directory for example `odata-client`.
+
+This will generate OData clients for all your service specifications, but also in the inputDir a new file is created: `serviceMapping.json`.
+
+Inside this file, the `servicePath` has to be defined manually. Depending on the service that you are using, your `serviceMapping.json` could look like this:
+
+```json
+{
+  "RCMCandidate": {
+    "directoryName": "sfo-data-service",
+    "servicePath": "/odata/v2",
+    "npmPackageName": "sfo-data-service"
+  }
+}
+```
```

Comment:
```suggestion
This will generate OData clients for all your service specifications and create a `serviceMapping.json` in your input directory. This file is used for generation and contains a mapping from the original file name to the following information:
* `directoryName` - the name of the subdirectory the client code will be generated to.
* `servicePath` - the url path to be prepended before every request. This is read from the EDMX file if available, otherwise the value here will be `VALUE_IS_UNDEFINED`. In that case it should be adjusted manually.
* `npmPackageName` - the name of the npm package, if a package json is generated. This information is optional.

These information can be adjusted manually and ensure that every run of the generator produces the same names for the generation.

Example:
```json
{
  "MyService": {
    "directoryName": "my-service",
    "servicePath": "/odata/v2",
    "npmPackageName": "my-service"
  }
}
```

```